src/js/webrequest.js

```diff
@@ -325,6 +325,65 @@ function blockMozCspReports(details) {
   }
 }
 
+/**
+ * A backstop to remove undesirable redirects for when our JS approach fails.
+ *
+ * Some Google properties (such as Google Docs in Firefox) rewrite clicked
+ * link actions via some mix of capturing all events, window.open() with
+ * document.write(), and meta refresh.
+ *
+ * @param {Object} details webRequest request details object
+ *
+ * @returns {Object|undefined} Can redirect requests
+ */
+function bypassGoogleRedirects(details) {
+  if (!badger.INITIALIZED) { return; }
+  let request_host = extractHostFromURL(details.url);
+  if (!badger.isPrivacyBadgerEnabled(request_host)) { return; }
+  let urlObj, redirect_url;
```

Comment:
Suggesting to collapse these statements since they both lead to return on `false`.

```suggestion
  let request_host = extractHostFromURL(details.url);
  let urlObj, redirect_url;
  if (!badger.INITIALIZED || !badger.isPrivacyBadgerEnabled(request_host)) {
    return;
  }
```

Reply:
**Note**: I did not locally evaluate this code. So don't merging here may cause an error. The same goes for the rest of ny comments with code.

Reply:
It's important to check `badger.INITIALIZED` before anything else happens as event listeners could be called before Privacy Badger's various internals are ready to respond. Also here we don't need to do the work of extracting the host from URL if Badger is not yet initialized.

Reply:
Is there a global way to check `badger.INITIALIZED`? Just noticing you have to check in function a few times.

Reply:
There are a bunch of event listeners, and most of them check `badger.INITIALIZED` (I think there are a few that don't have to).

We need to register event listeners ASAP (in the first turn of the event loop), but we don't want to generate errors when the listeners get called before our various async-init data structures are ready.

We could create a layer of indirection, perhaps a utility function that creates event listener functions. That could certainly be an improvement in the future; I intentionally postpone adding layers as much as possible.